akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1435,6 +1435,7 @@ private[stream] object Collect {
     private var groupEmitted = true
     private var finished = false
     private var totalWeight = 0L
+    private var elementCount = 0L
```

Comment:
in `onTimer` instead of:
```
if (elementCount > 0L) {
  if (isAvailable(out)) emitGroup()
  else pushEagerly = true
}
```
what should I do?
if there are elements, but `isAvailable(out)==false`, we just set `pushEagerly = true` and continue to buffer elements until downstream pulls, which then we immediately emit all buffered elements. I can't call `.result()` since I won't be able to buffer more elements if I can't push downstream right now.

maybe something like:
```
def onTimer(timerKey: Any) = if (isAvailable(out)) {
  val elems = buf.result()
  buf.clear()
  if (elems.nonEmpty) emitGroup(elems)
} else pushEagerly = true
```

but it changes the behavior.  
If no elements were buffered at all, it would set `pushEagerly = true`,
which mean that next `onPull()` will cause eager `push`, even if no elements were buffered at all.
So we emit empty sequences...

Or we change the stage behavior even more.
seems like a too big change IMO, just to prevent an edge case...